### ISSUE TITLE
Using with very large images

### ISSUE THREAD

**Issue by arielctal** [nicinabox/superslides#36]
Hi,
I'm trying to use your library with very large images - slideshow of pictures directly from a camera. The result is that the images are not fitted to the screen but cropped at the original size.

What can I do to avoid that?

Thanks!

**Comment by nicinabox** [comment]
Good question. Images are set so the min-width is 100% of the image size. In other words, the image will not scale smaller than it's natural dimensions. It's never really come up before and the images that I've always used have been ~1000px sized for the web. 

Maybe it would be good to have a configurable option for this.

**Comment by randalldon** [comment]
Hi there,

Is there any way of correcting this? My client has requested that i use large resolution images but currently the behavoir is as arielt1985 suggested. I really need to be able to use larger resolution images and allow these to be scaled down in the same fashion.

Is there any update to the js that can be made to address this?

It's a bit of an urgent request really

Cheers

**Comment by nicinabox** [comment]
I'll have a look this morning.

**Comment by randalldon** [comment]
Hi there,

Did you manage to have a look at this?

Cheers

**Comment by markyuk** [comment]
Just come here looking for a solution the same issue - wanting high res images for retina and large screen displays but it's currently not sizing down :-(

**Comment by nicinabox** [comment]
Currently in progress on this.

**Comment by markyuk** [comment]
Ha, amazingly fast reply! :-)

Question for you, if I may - I need to get this working TODAY .... could I set max-width:100% and min-height:100% after loaded?

I can gladly send you a sample URL (email is mark at [logicspot] .._DOT_.. com)

**Comment by nicinabox** [comment]
@markyuk if all your images are larger than the viewport, yes that will work. `min-height` is already set on the image. You'd need to remove `min-width` and replace with `max-width`.

**Comment by markyuk** [comment]
I tried this ... but when the viewport is LONGER than wider, the image is STRETCHED in the height :-(

**Comment by randalldon** [comment]
Same issue here.

**Comment by nicinabox** [comment]
Yeah I see that. Bummer, I thought that might work. I'm adding some checks now to better scale the image.

**Comment by nicinabox** [comment]
I think I've got it working :)

**Comment by nicinabox** [comment]
@markyuk, @randalldon Can you grab the code from master branch to test for me?

**Comment by randalldon** [comment]
Looks all good. I've tested from the master in Chrome 24 Firefox 18, Safari 5.1, ie7, 8 & 9 on PC. Could you let me know which files have been updated?

Cheers

**Comment by nicinabox** [comment]
Sweet, thanks! You'll probably want to compare what's in dist/ to what you've got. More than likely you'll just want to grab the javascript file.

Also just fixed #38 and #40.

**Comment by markyuk** [comment]
Thanks there ... works brilliantly here :-)  Awesome turnaround dude .... thanks

**Comment by randalldon** [comment]
Agreed. Great work and thanks for looking into it so quickly :+1:

**Comment by nicinabox** [comment]
And thank _you_ for your feedback :)

**Comment by randalldon** [comment]
Hi there,

Update to testing on this.

There does indeed seem to be issues with ie 7 & 8 now. I'm not seeing the image scaling at all in these browsers. For me the image is always it's natural dimension size and doesn't scale either up or down as the window is resized. Can you confirm this? ie 9 appears fine as does Firefox, Chrome & Safari though.

Cheers

**Comment by nicinabox** [comment]
I'm not seeing any issues with IE 7 or 8 as far as image resizing goes. IE7 seems to not want to cycle through the slides though, at least in the demo.

**Comment by randalldon** [comment]
Which demo link are you referencing?

**Comment by nicinabox** [comment]
The [marketing demo](http://nicinabox.github.com/superslides/#0).

**Comment by randalldon** [comment]
This demo doesn't use any large resolution images though. They're all 1024x683.

**Comment by nicinabox** [comment]
DOH! Right, scatterbrained this morning; will check again.

**Comment by randalldon** [comment]
Just to let you know the behavior i'm seeing it seems to be the same for both ie7 & ie 8.
1. I'm viewing my demo site at full hd so 1920 x 1080 resolution.
2. Some images are 1400x875 and some are larger.
3. This shot shows the effect of a smaller image in ie7 & ie8 - http://s1077.beta.photobucket.com/user/randalldon/media/superslides-test-shot1_zps4704bfa3.jpg.html i.e. image is never scalable and always presented at it's original dimensions.
4. This shows an initial load of the larger image which is fine in both ie7 & ie8 - http://s1077.beta.photobucket.com/user/randalldon/media/superslides-test-shot2_zps16537811.jpg.html?sort=3&o=1 - teh image is scalable at this point.
5. However, in both ie7 & ie 8 if i navigate away from the page and then come back to it this is the result - http://s1077.beta.photobucket.com/user/randalldon/media/superslides-test-shot3_zpsce40b3fd.jpg.html?sort=3&o=0 i.e. the image goes back to it's original dimensions and doesn't scale any more. A hard refresh resets to the image to the state from point 4 but it loses it again as soon as you navigate away and return.

As i mentioned this behavior only seems to apply to ie7 & ie8 with Safari, Chrome, Firefox all showing the correct behavior i.e. the image always fills the screen and is scalable.

Hope this helps

Cheers

**Comment by randalldon** [comment]
Just to add to this, and to eliminate any issue with my code i pulled down the master branch and placed a two test images in the index.html file with /demo. The behavior is the same as detailed above in ie7 & ie8 but fine again for other browsers.

Cheers

**Comment by nicinabox** [comment]
This is very helpful, thanks!

**Comment by randalldon** [comment]
Hey There,

Any joy with looking into this?

Cheers

**Comment by nicinabox** [comment]
I haven't yet had time in my production schedule. I only have access to a windows machine at work, so I have to work it in sometime during the workday (maybe today!).

**Comment by randalldon** [comment]
No problem. It's a bit of a killer at the moment with images not functioning effectively for ie7 & ie8 :(.

**Comment by randalldon** [comment]
Hi There,

Sorry if i'm being a pain on this as i'm sure you're busy but i was just wondering if you've managed to look into this any further? It's the last piece of the jigsaw for me so to speak.

Cheers

**Comment by nicinabox** [comment]
You're not a pain! I appreciate the follow ups. I promise, I haven't forgotten about you :). Unfortunately, I've been on a rather tight production schedule last week and this week. That said, I don't think it will be difficult to fix, just need some time to sit down and thoroughly test.

**Comment by randalldon** [comment]
No problem. I really appreciate your efforts.

Cheers

**Comment by randalldon** [comment]
Hi There,

I don't suppose you've managed to take a look at this at all? I understand if you've not had the time.

Cheers

**Comment by randalldon** [comment]
Hi there,

I'm sure you're very busy but has the been any further investigation into this?

Cheers

**Comment by nicinabox** [comment]
@randalldon working on it today!

**Comment by nicinabox** [comment]
@randalldon If you don't mind testing this out for me... Grab the latest from 0.5-stable and see if it fixes your problem.

**Comment by randalldon** [comment]
Hi There,

I've pulled down the latest stable as you suggested to test through IE and it seems there's some strange issues. If the image is smaller than the viewport i.e i'm testing on full hd so 1920 x 1080 and i reference your 'people.jpeg' it seems to work fine. However, if i use an image which is larger than the initial viewport so something like 2560x1440 the image doesn't load at all in any version of IE. The missing image icon seems to appear and the page background is just blank (please see the attached screen-shot). This behavior is the same for ie7, 8 & 9.

Cheers
![superslides-ie-image-issues](https://f.cloud.github.com/assets/2835493/215983/98e9083e-84b0-11e2-8878-4d0519d340b3.jpg)

**Comment by nicinabox** [comment]
That seems different that what you experienced before? Not loading at all though, that's odd. I'll try with a bigger image.

**Comment by nicinabox** [comment]
Hm, just tried with an image at 2048x1420 and seemed to perform as expected. My viewport is 1024x768 (virtual machine). Even the 'people.jpg' is larger, and it seems to work as you reported. 

Are you sure it's not an isolated case with the image you tried to load?

**Comment by randalldon** [comment]
Hi There,

Thanks for coming back to me. I've tried with several images with the same result. The images worked in all browsers other than these ie versions. Can you confirm which version of Superslides has these fixes? If i pull down the stable version is that the one you're using to test?

Many thanks

**Comment by nicinabox** [comment]
Yep, 0.5-stable. Can you give me more info about your machine?
—
@nicinabox

**Comment by nicinabox** [comment]
Just tried on IE8 native and still not seeing it.

![ie-wide](https://f.cloud.github.com/assets/246280/232857/0a4a9814-8743-11e2-9ff6-57dfaf661a50.png)
![ie8-narrow](https://f.cloud.github.com/assets/246280/232858/0a557fea-8743-11e2-88db-19ffec12cf8b.png)
![ie8-fullscreen](https://f.cloud.github.com/assets/246280/232859/0a61b26a-8743-11e2-8e35-f5b3fc787ce1.png)
